src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,96 @@
+package duke.tasks;
+
+import java.io.Serializable;
+
+import java.security.InvalidParameterException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Superclass that supports task methods
+ */
+public class Task implements Serializable {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for task invoked by superclass
+     * @param description Description of task
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * If tasks is done indicates "X" else shows a blank
+     * @return returns a string of either X or blank
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Marks the task as done
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as undone
+     */
+    public void unmarkAsDone() { this.isDone = false; }
```

Comment:
Should this use the Egyptian-style brackets?

```suggestion
/**
 * Marks the task as undone
 */
public void unmarkAsDone() {
    this.isDone = false;
}
```